internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -931,6 +984,42 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return resourceRead(ctx, d, meta)
 }
 
+func handlePinnedFCVUpdate(ctx context.Context, connV2 *admin.APIClient, projectID, clusterName string, d *schema.ResourceData, timeout time.Duration) diag.Diagnostics {
+	if d.HasChange("pinned_fcv") {
+		if pinnedFCVBlock, ok := d.Get("pinned_fcv").([]any); ok && len(pinnedFCVBlock) > 0 {
+			if diags := pinFCV(ctx, connV2, projectID, clusterName, pinnedFCVBlock[0]); diags.HasError() {
+				return diags
+			}
+		} else {
+			// pinned_fcv has been removed from the config so unpin method is called
+			if _, _, err := connV2.ClustersApi.UnpinFeatureCompatibilityVersion(ctx, projectID, clusterName).Execute(); err != nil {
+				return diag.FromErr(fmt.Errorf(errorUpdate, clusterName, err))
+			}
+		}
+		// ensures cluster is in IDLE state before continuing with other changes
+		if err := waitForUpdateToFinish(ctx, connV2, projectID, clusterName, timeout); err != nil {
+			return diag.FromErr(fmt.Errorf(errorUpdate, clusterName, err))
+		}
+	}
+	return nil
+}
+
+func pinFCV(ctx context.Context, connV2 *admin.APIClient, projectID, clusterName string, fcvBlock any) diag.Diagnostics {
```

Comment:
[nit] Interesting choice passing the `fcvBlock any`, would consider passing `schema.ResourceData` instead and doing more logic here but no strong opinion

Reply:
in both create and update before calling pinFCV I am checking that the block is defined, so I can pass that info directly. Don't have a strong opinion but helps me understand that pinFCV is called having the block defined.